relay-server/src/actors/healthcheck.rs

```diff
@@ -17,16 +17,33 @@ use crate::statsd::RelayGauges;
 
 lazy_static::lazy_static! {
     /// Singleton of the `Healthcheck` service.
-    static ref ADDRESS: RwLock<Option<Addr<HealthcheckMessage>>> = RwLock::new(None);
+    static ref ADDRESS: RwLock<Option<Addr<Healthcheck>>> = RwLock::new(None);
 }
 
-/// Internal wrapper of a message sent through an `Addr` with return channel.
-#[derive(Debug)]
-struct Message<T> {
-    data: T,
-    // TODO(tobias): This is hard-coded to return `bool`.
-    // Might need some associated types to make this work
-    responder: oneshot::Sender<bool>,
+/// Our definition of a service.
+///
+/// Services are much like actors: they receive messages from an inbox and handles them one
+/// by one.  Services are free to concurrently process these messages or not, most probably
+/// should.
+///
+/// Messages always have a response which will be sent once the message is handled by the
+/// service.
+pub trait Service {
+    /// The envelope is what is sent to the inbox of this service.
+    ///
+    /// It is an enum of all the message types that can be handled by this service together
+    /// with the response [sender](oneshot::Sender) for each message.
+    type Envelope;
```

Comment:
At first, I was thinking to constrain the `Envelope` type with `Send + 'static`. Even single-threaded services would be called from other threads, so the `Envelope` most certainly must flow across thread boundaries. 

Annotating that here may improve compiler messages, but it's also not strictly necessary. I was thinking if there might even be a benefit to having non-`Send` services, that coexist in a single-threaded runtime.

(if you end up going with `Send + 'static` here, this would also apply to `ServiceMessage::Response`).

Reply:
i like `Send + 'static`